src/ansi-c/library/new.c

```diff
@@ -6,7 +6,13 @@ inline void *__new(__typeof__(sizeof(int)) malloc_size)
   // This just does memory allocation.
   __CPROVER_HIDE:;
   void *res;
-  res=__CPROVER_malloc(malloc_size);
+  // ensure that all bytes in the allocated memory can be addressed
+  // using our object:offset encoding as specified in
+  // flattening/pointer_logic.h; also avoid sign-extension issues
+  // for 32-bit systems that yields a maximum allocation of 2^23-1,
+  // i.e., just under 8MB
+  __CPROVER_assume(malloc_size<(1ULL<<((sizeof(char*)-1)*8-1)));
```

Comment:
Yes, that should be considered here (even though I'd prefer #1086 to eventually fix this properly), but needs some extra work to expose the configuration option to the C preprocessor. I should likely move this to a different PR.